src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DepartureReasonEntity.kt

```diff
@@ -34,14 +40,28 @@ data class DepartureReasonEntity(
   val isActive: Boolean,
   val serviceScope: String,
   val legacyDeliusReasonCode: String?,
+  val parentReasonId: UUID?,
 ) {
   override fun toString() = "DepartureReasonEntity:$id"
 
   companion object {
     fun valueOf(apiValue: WithdrawPlacementRequestReason): PlacementRequestWithdrawalReason? =
       PlacementRequestWithdrawalReason.entries.firstOrNull { it.apiValue == apiValue }
+
+    fun generateParentChildName(reason: DepartureReasonEntity?): NamedId? =
```

Comment:
i think it's probably more flexible if we

a) converted parentReasonId into a nullable @ManyToOne relationship to DepartureReasonEntity
b) move the generateParentChildName function into the class definition itself (instead of it being a 'static' function) and then follow this relationship if defined (no need to hardcode the BREACH RECALL id then)?